# Computational-Experimental-framework-for-predicting-effects-of-variants-on-alternative-splicing


Splicing is a highly regulated process that depends on numerous factors. It is particularly challenging to quantitatively predict how a mutation will affect precursor messenger RNA (mRNA) structure and the subsequent functional consequences. Here we use a novel Mutational Profiling (-MaP) methodology to obtain highly reproducible endogenous precursor and mature mRNA structural probing data in vivo. We use these data to estimate Boltzmann suboptimal ensembles, and predict the structural consequences of mutations on precursor mRNA structure. Together with a structural analysis of recent cryo-EM spliceosome structures at different stages of the splicing cycle, we determined that the footprint of the BAct complex on precursor mRNA is best able to predict splicing outcomes for exon 10 inclusion of the alternatively spliced MAPT gene. However, structure alone only achieves 74% accuracy. We therefore developed a ÔÅ¢-regression weighting framework that incorporates splice site strength, structure and exonic/intronic splicing regulatory elements which together achieves 90% accuracy for 47 known and six newly discovered splice-altering variants. This combined experimental/computational framework represents a path forward for accurate prediction of splicing related disease-causing variants.
